### PR TITLE
fix index overflow

### PR DESCRIPTION
**Pull request by yuval-k**

### CODE REVIEW

File: changelog/v0.11.3/index_overflow.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: FIX
+    description: something useful here
```
Comment (EItanya) [inline changelog/v0.11.3/index_overflow.yaml]: Comment description

File: projects/gloo/pkg/plugins/kubernetes/uds_convert_test.go [hunk 2]
```diff
@@ -24,6 +26,14 @@ var _ = Describe("UdsConvert", func() {
 		}
 		Expect(result).To(Equal(expected))
 	})
+	It("should truncate long names", func() {
+		name := UpstreamName(strings.Repeat("y", 120), "gloo-system", 12, nil)
+		Expect(name).To(HaveLen(63))
+	})
+	It("should truncate long names with long of labels", func() {
```
Comment (rickducott) [inline projects/gloo/pkg/plugins/kubernetes/uds_convert_test.go]: nit: lot of labels

File: projects/gloo/pkg/plugins/kubernetes/uds_convert.go [hunk 3]
```diff
@@ -138,7 +138,9 @@ func UpstreamName(serviceNamespace, serviceName string, servicePort int32, extra
 		hash := md5.Sum([]byte(name))
 		name = fmt.Sprintf("%s-%s-%v-%x", serviceNamespace, serviceName, servicePort, hash)
 		// todo: ilackarms: handle potential collisions
```
Comment (rickducott) [inline projects/gloo/pkg/plugins/kubernetes/uds_convert.go]: There was a comment here when this originally was written that having the hash at the end could cause collisions, especially if there's no limit on namespace name (see other comment)

File: projects/gloo/pkg/plugins/kubernetes/uds_convert_test.go [hunk 4]
```diff
@@ -24,6 +26,14 @@ var _ = Describe("UdsConvert", func() {
 		}
 		Expect(result).To(Equal(expected))
 	})
+	It("should truncate long names", func() {
+		name := UpstreamName(strings.Repeat("y", 120), "gloo-system", 12, nil)
```
Comment (rickducott) [inline projects/gloo/pkg/plugins/kubernetes/uds_convert_test.go]: this first arg is the namespace.. if we had two upstreams in this namespace they would not be uniquely identified. Maybe out of scope for this PR, but maybe worth filing ?

Comment (yuval-k) [reply projects/gloo/pkg/plugins/kubernetes/uds_convert_test.go]: I think it's easy enough to fix in this PR;

File: projects/gloo/pkg/plugins/kubernetes/uds_convert.go (hunk 3 above)
Comment (ilackarms) [reply projects/gloo/pkg/plugins/kubernetes/uds_convert.go]: truncating the hash can cause collisions

Comment (mitchdraft) [reply projects/gloo/pkg/plugins/kubernetes/uds_convert.go]: we should make a util for generating valid composite names, length limit on k8s names is `253`
"By convention, the names of Kubernetes resources should be up to maximum length of 253 characters and consist of lower case alphanumeric characters, -, and ., but certain resources have more specific restrictions."
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names

**Pull request merged**